backup/backup_client.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash                                                                                                                                                 
+
+set -euo pipefail
+
+if [ "${1:-}" = "backup-db" ] ; then
+    PGPASSWORD="${DB_PASS}" pg_dump -h "${DB_HOST}" -p "${DB_PORT}" -U "${DB_USER}"
+    exit
+fi
+
+caddy_paths=("/opt/caddy")
+
+mastodon_paths=("/opt/mastodon/public"
+                "/opt/mastodon/elasticsearch"
+                "/opt/mastodon/redis"
+                "/opt/mastodon/mastodon.env"
+                "/opt/mastodon/versions.env")
+
+monitoring_paths=("/opt/mastodon/grafana"
+                  "/opt/mastodon/prometheus"
+                  "/opt/mastodon/monitoring-versions.env"
+                  "/opt/mastodon/monitoring-admin.env"
+                  "/opt/mastodon/postgres-monitoring-queries.yaml")
+
+extra_paths=("/opt/backup"
+             "/opt/mastodon/wait-for-container.sh"
+             "/opt/mastodon/setup-mastodon.sh")
+
+service_wildcards=("/etc/systemd/system/mastodon*"
+                   "/etc/systemd/system/monitoring-*"
+                   "/etc/systemd/system/metrics-provider-*"
+                   "/etc/systemd/system/caddy*.service")
```

Comment:
Now that I'm reading it, I think that `shopt -s nullglob` should happen before `service_wildcards` are defined, and the wildcard expansion should happen here, because `"${service_wildcards[@]}"` does not expand the globs:
```suggestion
shopt -s nullglob
service_wildcards=("/etc/systemd/system/mastodon"*
                   "/etc/systemd/system/monitoring-"*
                   "/etc/systemd/system/metrics-provider-"*
                   "/etc/systemd/system/caddy"*".service")
```

Reply:
`"${service_wildcards[@]}"` actually broke things, I've changed it to `${service_wildcards[@]}`. After that change the wildcard expansion does happen at the `tar ...` line, and the `nullglob` line is correct.